api/core/helper/code_executor/code_executor.py

```diff
@@ -112,6 +112,9 @@ def execute_code(cls,
 
         if response.code != 0:
             raise CodeExecutionException(response.message)
+
+        if response.data is None:
+            raise CodeExecutionException('Failed to execute code, got empty response')
```

Comment:
`CodeExecutionResponse.data` is still marked as Optional, which means we cannot be sure if it is a NoneType before using it. Currently, because `CodeExecutionResponse` is only constructed in this one place, we can be certain it won't be a NoneType. However, this is an implicit agreement that might cause issues in the future.

Regarding the issue mentioned in #6829, I suggest checking the value of `code` before constructing `CodeExecutionResponse`. How about modify from L112:

```python3
        if (code := response.get('code', -1)) != 0:
            raise CodeExecutionException(response.get('message', f'failed with {code=}'))
        response = CodeExecutionResponse(**response)
```

Then keep `CodeExecutionResponse.data` as `Data`.

Reply:
It is a good idea!  Here I don't give default `-1` to `code`, maybe None value is help us more to debug